components/MapFilters.vue

```diff
@@ -74,171 +107,204 @@
      },
-     getMarkerIcon: function (marker) {
-       // Sets the incon of a marker depending on its
-       // marker type, which it gets from calling
-       // getMarkerType.
+     getMarkerState: function (marker) {
+       // Returns a marker state array [markerIcon, onOff] based
+       // on the witches that are on. It gets the markerType by calling
+       // getMarkerTypeOnFilter, and returns the state accordingly.
+       // Note that if getMarkerTypeOnFilter returns null, then no witches
+       // are on so the marker is off.
+       let markerType = this.getMarkerType(marker.witches);
 
-       let markerType = this.getMarkerType(marker);
-
-       if (markerType === 'mixed') {
-         return '/images/witch-single-purple.png';
-       } else {
-         return this.filterLayers[this.currentLayer].filters[markerType].iconUrl;
+       if (markerType) {
+         return [this.getMarkerIcon(markerType), true];
        }
+       return [null, false];
      },
-     updateMarkerState: function (marker) {
-       // Updates the state of a marker after it being
-       // filtered. Specifically, its marker Icon and
-       // if it needs to be on or off depending on whether
-       // all the witches have been filtered out.
-
-       if (marker.witches.length > 0) {
-         marker.markerIcon = this.getMarkerIcon(marker);
-         marker.onOff = true;
-       } else {
-         marker.onOff = false;
+     getIsMixed: function (marker) {
+       // Checks whether a marker is mixed.
+       if (marker.markerIcon === '/images/witch-single-purple.png') {
+         return true;
        }
-
-       return marker;
+       return false;
      },
-     removeWitchesFromArray: function (witches, filterType) {
-       // Given an array of witches, it filters out the
-       // witches that meet the property filterType.
+     setWitchesOff: function (filterProperty, filterType) {
+       // Filters <filterProperty>.<filterType> off. It goes through the current markers
+       // setting witches that meet filter type to off and adds the
+       // filter property to the witche's active filters.
+       // If a marker is mixed, meaning it could stop being mixed, it
+       // updates the marker state by calling getMarkerState.
 
-       let filteredWitches = [];
+       for (let i = 0, len = this.markers.length; i < len; i++) {
+         let marker = this.markers[i];
 
-       for (let i = 0, len = witches.length; i < len; i++) {
-         let witch = witches[i]
+         for (let w = 0, len = marker.witches.length; w < len; w++) {
+           let witch = marker.witches[w];
 
-         if (witch[this.currentFilterProperty] !== filterType) {
-           filteredWitches.push(witch);
+           if (witch[filterProperty] === filterType) {
+             witch.witchState.onOff = false;
+             witch.witchState.activeFilters.push(filterProperty);
+           }
          }
+         [marker.markerIcon, marker.onOff] = this.getMarkerState(marker);
```

Comment:
Please add new line above [marker.markerIcon, marker.onOff] = this.getMarkerState(marker);

For control structures, our team add a new line above and below for readability. e.g.

```
let x =1;
for(let i = 0; i < 10; i++) {  
    console.log(i);
}
let y = 5;
```

Should be:

```
let x =1;

for(let i = 0; i < 10; i++) {  
    console.log(i);
}

let y = 5;

```
Same for if statements

Reply:
Resolved.